src/mlpack/methods/reinforcement_learning/q_learning_impl.hpp

```diff
@@ -48,9 +48,14 @@ QLearning<
     totalSteps(0),
     deterministic(false)
 {
+  // To copy over the network structure.
+  targetNetwork = learningNetwork;
+
   // Set up q-learning network.
   if (learningNetwork.Parameters().is_empty())
     learningNetwork.ResetParameters();
+  if (targetNetwork.Parameters().is_empty())
```

Comment:
I would probably always do a reset here, just to be safe.

Reply:
Yeah, its better we do that.